hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1259,10 +1280,24 @@ public ListKeysResult listKeys(String volumeName, String bucketName,
           break;
         }
       }
+      readFromRDbStopNs = Time.monotonicNowNanos();
     }
 
     boolean isTruncated = cacheKeyMap.size() > maxKeys;
 
+    if (perfMetrics != null) {
+      long averagePagination;
+      if (isTruncated) {
+        averagePagination = maxKeys;
+      } else {
+        averagePagination = cacheKeyMap.size();
+      }
+      perfMetrics.setListKeysAveragePagination(averagePagination);
+      long opsPerNs =
+          averagePagination / (Time.monotonicNowNanos() - startNanos);
+      perfMetrics.setListKeysOpsPerNs(opsPerNs);
```

Comment:
Will there ever by more than 0 ops per nanos, given integer division rounds down?

Reply:
@adoroszlai, sure changed it.